crates/matrix-sdk-base/src/store/observable_map.rs

```diff
@@ -249,6 +272,30 @@ mod tests {
         assert_eq!(map.get(&'c'), Some(&'G'));
     }
 
+    #[test]
+    fn test_remove() {
+        let mut map = ObservableMap::<char, char>::new();
+
+        assert!(map.get(&'a').is_none());
+        assert!(map.get(&'b').is_none());
+        assert!(map.get(&'c').is_none());
+
+        // new items
+        map.insert('a', 'e');
+        map.insert('b', 'f');
+
+        assert_eq!(map.get(&'a'), Some(&'e'));
+        assert_eq!(map.get(&'b'), Some(&'f'));
+        assert!(map.get(&'c').is_none());
+
+        // remove one item
+        map.remove(&'b');
+
+        assert_eq!(map.get(&'a'), Some(&'e'));
+        assert_eq!(map.get(&'b'), None);
+        assert_eq!(map.get(&'c'), None);
```

Comment:
Can you also `map.remove(&'d')` —which doesn't exist— to assert it returns `None` please?

Reply:
Ah good catch!

Reply:
Nice, thanks!